### PR TITLE
Assembly hub support

### PR DESCRIPTION
**Pull request by Maarten-vd-Sande**
See #172 

**What problem is the PR solving / What's new?**
WIP

**What did change**
WIP

**Does this need additional test(s)?**
WIP

### CODE REVIEW

**Comment by Maarten-vd-Sande** [comment]
@siebrenf, what are the odds that the assembly is not supported by ucsc, but an annotation exists? Did you come across a situation like that?

**Comment by Maarten-vd-Sande** [comment]
Something is wrong with the tests, it does not test everything :unamused:

**Comment by Maarten-vd-Sande** [comment]
@Maarten-vd-Sande add mkdir to trackhub, reported that this sometimes causes issues?

File: rules/trackhub.smk [hunk 1]
```diff
@@ -96,38 +99,169 @@ rule narrowpeak_bignarrowpeak:
...
+                                allow_redirects=True)
+        assert response.ok, "Make sure you are connected to the internet"
+
+        # see if the title of the page mentions our assembly
+        if not any(assembly == x for x in
+                   re.search(r'<TITLE>(.*?)</TITLE>', response.text).group(1).split(' ')):
+            trackfiles['twobits'].append(f"{config['genome_dir']}/{assembly}/{assembly}.2bit")
+
+        # now check if we have annotation for
+        # TODO
```
Comment (siebrenf) [inline rules/trackhub.smk]: for later?

File: envs/ucsc.yaml [hunk 2]
```diff
@@ -7,3 +7,4 @@ dependencies:
   - bioconda::ucsc-bedsort=377
   - bioconda::ucsc-bedGraphToBigWig=377
   - bioconda::ucsc-bedToBigBed=377
+  - bioconda::ucsc-fatotwobit=377
```
Comment (siebrenf) [inline envs/ucsc.yaml]: ucsc.yaml is for the track-/assembly hub? To me, the purpose is unclear from the name.

File: rules/trackhub.smk (hunk 1 above)
Comment (Maarten-vd-Sande) [reply rules/trackhub.smk]: Yes, its waiting for the annotation downloads and a bugfix from Snakemake (which they actually fixed last week :D). 

That part is necessary for nice assembly hub with genes, but not essential for assembly hub functionality

File: envs/ucsc.yaml (hunk 2 above)
Comment (Maarten-vd-Sande) [reply envs/ucsc.yaml]: All the ucsc related tools are there, not the dependencies for the trackhub/assembly hub. Feel free to change it to sth more clear

**Pull request merged**